test/services/filterServiceTest.js

```diff
@@ -95,7 +95,7 @@ describe('FilterService', function () {
     const filterWithEventTypesAndStartEndTimes = {startTime: 'startTime', endTime: 'endTime', eventTypes: [1, 2]}
     const filterWithNoEventTypes = {startTime: 'startTime', endTime: 'endTime'}
 
-    const expectedResult = '/events/?eventTypes=1&eventTypes=2&startTime=startTime&endTime=endTime'
+    const expectedResult = '/events/?startTime=startTime&endTime=endTime&eventTypes=1&eventTypes=2'
```

Comment:
Rather than changing the order of the expected result string, change your expectations to not require order-dependancy by using `include`. The expected result could be an array of parameter strings

``` javascript
const expectedResult =  [ 'eventTypes=1', 'eventTypes=2', 'startTime=startTime', 'endTime=endTime' ]
```

Then alter the expectation to read something like:

``` javascript
expect(history[0]).to.include(expectedResult)
```

Reply:
include only works between arrays. history[0] is a string

Reply:
[String also has include](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes), and the [Chai api supports it](http://www.chaijs.com/api/bdd/#method_include)

```javascript
const expectedResult =  [ 'eventTypes=1', 'eventTypes=2', 'startTime=startTime', 'endTime=endTime' ]

expectedResult.forEach(expectedToken => expect(history[0]).to.include(expectedToken))
```

Could work.